Most popular transport types:
What are the most popular transport types, measured by the total number of journeys?

In [22]:
SELECT journey_type, ROUND(SUM(journeys_millions),2) AS TotalJourneysMillion
FROM TFL.JOURNEYS
GROUP BY journey_type
ORDER BY TotalJourneysMillion DESC

journey_type  TotalJourneysMillion
0                Bus              24905.19
1  Underground & DLR              15020.47
2         Overground               1666.85
3           TfL Rail                411.31
4               Tram                314.69
5   Emirates Airline                 14.58

The data reveals that bus journeys significantly lead with 24,905.19 million, indicating their crucial role in the transport network. Underground & DLR follow with 15,020.47 million, highlighting their importance as well. Overground and TfL Rail have moderate usage at 1,666.85 million and 411.31 million respectively, while Tram and Emirates Airline show niche utilization with 314.69 million and 14.5 million journeys. This distribution underscores the varied reliance on different transport modes within the network.

In [7]:
Select tl.year, ROUND(SUM(journeys_millions),3) AS Journeys_Volume
From TFL.JOURNEYS tl
GROUP BY tl.year
Order by tl.year DESC

year  Journeys_Volume
0   2022         2995.517
1   2021         2283.443
2   2020         1186.591
3   2019         3633.526
4   2018         3672.715
5   2017         3678.436
6   2016         3714.685
7   2015         3752.506
8   2014         3626.799
9   2013         3580.344
10  2012         3517.051
11  2011         3429.604
12  2010         3261.877

The number of journeys (in millions) peaked in 2015, followed by a decline in 2017. After three stable years, the most significant drop occurred in 2020. 

Although there was an increase in 2021 and 2022, the total journeys remain 17% below the 2019 level.

How is the trend in year for different types of transports?

In [2]:
SELECT year, ROUND(SUM(journeys_millions), 2) AS total_journeys_millions
FROM TFL.JOURNEYS
WHERE journey_type = 'Bus'
GROUP BY year
ORDER BY year DESC

year  total_journeys_millions
0   2022                  1642.49
1   2021                  1363.96
2   2020                   792.07
3   2019                  1989.85
4   2018                  2037.66
5   2017                  2070.91
6   2016                  2083.44
7   2015                  2154.71
8   2014                  2177.69
9   2013                  2175.57
10  2012                  2148.50
11  2011                  2153.29
12  2010                  2115.05

In [3]:
SELECT year, ROUND(SUM(journeys_millions), 2) AS total_journeys_millions
FROM TFL.JOURNEYS
WHERE journey_type = 'Underground & DLR'
GROUP BY year
ORDER BY year DESC

year  total_journeys_millions
0   2022                  1064.86
1   2021                   748.45
2   2020                   310.18
3   2019                  1386.44
4   2018                  1382.42
5   2017                  1362.29
6   2016                  1384.64
7   2015                  1363.46
8   2014                  1289.36
9   2013                  1250.43
10  2012                  1225.15
11  2011                  1156.65
12  2010                  1096.15

In [4]:
SELECT year, ROUND(SUM(journeys_millions), 2) AS total_journeys_millions
FROM TFL.JOURNEYS
WHERE journey_type = 'Overground'
GROUP BY year
ORDER BY year DESC

year  total_journeys_millions
0   2022                   143.87
1   2021                   117.08
2   2020                    54.70
3   2019                   174.15
4   2018                   173.89
5   2017                   174.98
6   2016                   173.72
7   2015                   170.54
8   2014                   127.22
9   2013                   124.46
10  2012                   113.82
11  2011                    93.47
12  2010                    24.95

Analysis of transport data shows different trends for underground, overground, and bus services. Underground and overground transport have remained steady or slightly increased year-over-year, except for a dip in 2020 due to COVID-19 restrictions.

Bus ridership, however, has been declining since 2015. This decline steepened in 2020 and, despite some recovery afterwards, has not returned to previous levels.

To understand why bus use is decreasing, we need to examine factors like delays, routes, costs, and customer satisfaction. This data would help explain the trends and guide improvements in bus services.

Which are the busiest months? Do we have specific months with heavier number of journeys?

In [10]:
SELECT month, ROUND(SUM(journeys_millions), 2) AS total_journeys
FROM TFL.JOURNEYS
GROUP BY month
ORDER BY month 

month  total_journeys
0       1         3416.60
1       2         3500.52
2       3         3479.68
3       4         3627.31
4       5         3302.09
5       6         3452.74
6       7         3840.90
7       8         3750.63
8       9         3847.42
9      10         2970.15
10     11         3557.84
11     12         3587.20

In [9]:
SELECT month, ROUND(AVG(journeys_millions), 2) AS total_journeys
FROM TFL.JOURNEYS
GROUP BY month
ORDER BY month 

month  total_journeys
0       1           49.52
1       2           51.48
2       3           50.43
3       4           51.82
4       5           47.17
5       6           49.32
6       7           54.87
7       8           52.83
8       9           54.19
9      10           41.83
10     11           50.11
11     12           50.52

We do have some variation however no month stands out as having significantly more or fewer journeys than the others.

Taking a look a the central metrics, the variability and distribution of the data.

In [45]:
SELECT journey_type, AVG(journeys_millions) AS mean_journeys
FROM TFL.JOURNEYS
GROUP BY journey_type
Order by mean_journeys DESC;

journey_type  mean_journeys
0                Bus     159.648679
1  Underground & DLR      96.285042
2         Overground      11.186884
3           TfL Rail       4.329615
4               Tram       2.017243
5   Emirates Airline       0.113052

In [46]:
SELECT journey_type,
       APPROX_QUANTILES(journeys_millions, 2)[OFFSET(1)] AS median_journeys
FROM TFL.JOURNEYS
GROUP BY journey_type
Order by median_journeys DESC;

journey_type  median_journeys
0                Bus       173.888609
1  Underground & DLR       102.628480
2         Overground        11.087398
3           TfL Rail         3.734558
4               Tram         2.166370
5   Emirates Airline         0.112039

The analysis of TfL journey data reveals a left-skewed, positive distribution where the median is greater than the mean for journey counts across different journey types.

This distribution pattern provides several insights:
- Most journey counts are clustered towards the higher end of the range, indicating that many journey types have relatively high usage.

Some journey types likely have significantly lower usage, pulling down the mean without greatly affecting the median.

- The median journey count (typical usage) is higher than the mean (average usage), suggesting that focusing on average figures alone might underestimate the typical journey volumes.

- There's greater variability in less-used journey types, while popular routes or types show more consistent usage patterns.

In [41]:
SELECT journey_type, ROUND(STDDEV(journeys_millions),2) AS stddev_journeys
FROM TFL.JOURNEYS
GROUP BY journey_type;

journey_type  stddev_journeys
0  Underground & DLR            26.40
1                Bus            35.77
2               Tram             0.46
3         Overground             3.54
4   Emirates Airline             0.06
5           TfL Rail             2.73

Bus and Underground & DLR networks exhibit the highest standard deviations in journey volumes, at 35.77 and 26.40 million journeys respectively. These substantial dispersions from the mean indicate significant volatility in ridership. As London's primary mass transit arteries, these services experience pronounced fluctuations, likely due to:

- Diurnal patterns (peak vs. off-peak hours).
- Hebdomadal cycles (weekday vs. weekend usage).
- Seasonal variations (holiday periods, academic calendars).
- Special events (concerts, sporting events, cultural festivities).

This high variability underscores the need for dynamic capacity management and adaptive scheduling strategies.

In [40]:
SELECT journey_type, ROUND(VARIANCE(journeys_millions),2) AS variance_journeys
FROM TFL.JOURNEYS
GROUP BY journey_type;

journey_type  variance_journeys
0  Underground & DLR             696.86
1                Bus            1279.63
2               Tram               0.21
3         Overground              12.53
4   Emirates Airline               0.00
5           TfL Rail               7.44

In [16]:
SELECT 
    journey_type,
    VARIANCE(journeys_millions) AS variance_journeys,
    CASE 
        WHEN VARIANCE(journeys_millions) > 100 THEN 'High Variance'
        WHEN VARIANCE(journeys_millions) BETWEEN 10 AND 100 THEN 'Medium Variance'
        ELSE 'Low Variance'
    END AS variance_category
FROM TFL.JOURNEYS
GROUP BY journey_type
ORDER BY variance_journeys DESC;

journey_type  variance_journeys variance_category
0                Bus        1279.628073     High Variance
1  Underground & DLR         696.863471     High Variance
2         Overground          12.528792   Medium Variance
3           TfL Rail           7.444215      Low Variance
4               Tram           0.209130      Low Variance
5   Emirates Airline           0.003751      Low Variance

In [15]:
SELECT 
    journey_type,
    AVG(journeys_millions) AS mean_journeys,
    VARIANCE(journeys_millions) AS variance_journeys,
    SQRT(VARIANCE(journeys_millions)) / AVG(journeys_millions) AS coefficient_of_variation
FROM TFL.JOURNEYS
GROUP BY journey_type
ORDER BY coefficient_of_variation DESC;

journey_type  ...  coefficient_of_variation
0           TfL Rail  ...                  0.630174
1   Emirates Airline  ...                  0.541723
2         Overground  ...                  0.316407
3  Underground & DLR  ...                  0.274167
4               Tram  ...                  0.226699
5                Bus  ...                  0.224066

[6 rows x 4 columns]

TfL Rail shows the highest relative variability, suggesting less predictable ridership patterns.
Emirates Airline, despite low absolute numbers, has high relative variability, possibly due to its niche service nature.
Bus and Tram services show the lowest relative variability, indicating more consistent ridership relative to their mean.

Tram and Bus services offer the most predictable ridership patterns relative to their mean usage.
Emirates Airline and TfL Rail may require more robust demand forecasting methods.


How has the variance been trending year after year for each of the journey types?

In [18]:
SELECT 
    year,
    journey_type,
    VARIANCE(journeys_millions) AS yearly_variance
FROM TFL.JOURNEYS
Where journey_type='Bus'
GROUP BY year, journey_type
ORDER BY year, yearly_variance DESC;

year journey_type  yearly_variance
0   2010          Bus       187.007592
1   2011          Bus       147.173626
2   2012          Bus       136.138387
3   2013          Bus       144.501077
4   2014          Bus       178.205640
5   2015          Bus       204.317536
6   2016          Bus       160.252025
7   2017          Bus       152.065201
8   2018          Bus       140.895014
9   2019          Bus       140.924639
10  2020          Bus       580.569200
11  2021          Bus       186.686958
12  2022          Bus        82.384067

In [30]:
SELECT 
    year,
    ROUND(AVG(journeys_millions),2) AS mean_journeys,
    ROUND(VARIANCE(journeys_millions),2) AS yearly_variance,
    ROUND(SQRT(VARIANCE(journeys_millions)) / AVG(journeys_millions),2) AS coefficient_of_variation
FROM TFL.JOURNEYS
WHERE journey_type = 'Bus'
GROUP BY year
ORDER BY year;

year  mean_journeys  yearly_variance  coefficient_of_variation
0   2010         176.25           187.01                      0.08
1   2011         179.44           147.17                      0.07
2   2012         179.04           136.14                      0.07
3   2013         181.30           144.50                      0.07
4   2014         181.47           178.21                      0.07
5   2015         179.56           204.32                      0.08
6   2016         173.62           160.25                      0.07
7   2017         172.58           152.07                      0.07
8   2018         169.81           140.90                      0.07
9   2019         165.82           140.92                      0.07
10  2020          66.01           580.57                      0.37
11  2021         113.66           186.69                      0.12
12  2022         136.87            82.38                      0.07

Bus ridership variability remained stable from 2010 to 2019, with a significant spike in 2020 due to the pandemic. Recent years show a trend towards pre-pandemic stability levels, indicating a gradual recovery in ridership patterns.

In [26]:
SELECT 
    year,
    ROUND(AVG(journeys_millions), 2) AS mean_journeys,
    ROUND(VARIANCE(journeys_millions), 2) AS yearly_variance,
    ROUND(SQRT(VARIANCE(journeys_millions)) / AVG(journeys_millions), 2) AS coefficient_of_variation
FROM TFL.JOURNEYS
WHERE journey_type = 'Underground & DLR'
GROUP BY year, journey_type
ORDER BY year;

year  mean_journeys  yearly_variance  coefficient_of_variation
0   2010          91.35            31.66                      0.06
1   2011          96.39            29.86                      0.06
2   2012         102.10            58.45                      0.07
3   2013         104.20            43.16                      0.06
4   2014         107.45            69.44                      0.08
5   2015         113.62            88.83                      0.08
6   2016         115.39            69.65                      0.07
7   2017         113.52            60.06                      0.07
8   2018         115.20            59.98                      0.07
9   2019         115.54            59.66                      0.07
10  2020          25.85           153.18                      0.48
11  2021          62.37           182.55                      0.22
12  2022          88.74            71.17                      0.10

Underground & DLR ridership variability was stable from 2010-2019, spiked dramatically in 2020 due to the pandemic, and has since been trending back towards pre-pandemic levels, indicating an ongoing recovery in ridership patterns.

In [36]:
SELECT 
    year,
    ROUND(AVG(journeys_millions), 2) AS mean_journeys,
    ROUND(VARIANCE(journeys_millions), 2) AS yearly_variance,
    ROUND(SQRT(VARIANCE(journeys_millions)) / AVG(journeys_millions), 2) AS coefficient_of_variation
FROM TFL.JOURNEYS
WHERE journey_type = 'Overground'
GROUP BY year, journey_type
ORDER BY year;

year  mean_journeys  yearly_variance  coefficient_of_variation
0   2010           4.99             0.74                      0.17
1   2011           7.79             1.00                      0.13
2   2012           9.48             0.71                      0.09
3   2013          10.37             0.71                      0.08
4   2014          10.60             0.55                      0.07
5   2015          14.21             2.14                      0.10
6   2016          14.48             1.09                      0.07
7   2017          14.58             1.12                      0.07
8   2018          14.49             1.21                      0.08
9   2019          14.51             3.71                      0.13
10  2020           4.56             5.32                      0.51
11  2021           9.76             3.50                      0.19
12  2022          11.99             3.74                      0.16

The coefficient of variation for Overground journeys was stable and low from 2010 to 2019, indicating consistent ridership. However, post-pandemic levels in 2022 remain higher than pre-pandemic, suggesting ongoing variability and a need for adaptive strategies to manage fluctuating demand.

In [35]:
SELECT 
    year,
    ROUND(AVG(journeys_millions), 2) AS mean_journeys,
    ROUND(VARIANCE(journeys_millions), 2) AS yearly_variance,
    ROUND(SQRT(VARIANCE(journeys_millions)) / AVG(journeys_millions), 2) AS coefficient_of_variation
FROM TFL.JOURNEYS
WHERE journey_type = 'TfL Rail'
GROUP BY year, journey_type
ORDER BY year;

year  mean_journeys  yearly_variance  coefficient_of_variation
0   2010            NaN              NaN                       NaN
1   2011            NaN              NaN                       NaN
2   2012            NaN              NaN                       NaN
3   2013            NaN              NaN                       NaN
4   2014           2.85              NaN                       NaN
5   2015           3.72             0.16                      0.11
6   2016           3.69             0.04                      0.06
7   2017           3.50             0.08                      0.08
8   2018           4.25             0.16                      0.09
9   2019           4.69             0.20                      0.09
10  2020           1.56             0.38                      0.40
11  2021           2.94             0.26                      0.17
12  2022          10.30            11.08                      0.32

From 2015 to 2019, TfL Rail showed low variability with a stable coefficient of variation. However, there was a sharp increase in variability during the pandemic in 2020, and while it decreased in 2021, the CV remained elevated in 2022 due to a significant rise in mean journeys, indicating ongoing fluctuations and potential demand shifts.

In [33]:
SELECT 
    year,
    ROUND(AVG(journeys_millions), 2) AS mean_journeys,
    ROUND(VARIANCE(journeys_millions), 2) AS yearly_variance,
    ROUND(SQRT(VARIANCE(journeys_millions)) / AVG(journeys_millions), 2) AS coefficient_of_variation
FROM TFL.JOURNEYS
WHERE journey_type = 'Tram'
GROUP BY year, journey_type
ORDER BY year;

year  mean_journeys  yearly_variance  coefficient_of_variation
0   2010           2.14             0.02                      0.06
1   2011           2.18             0.02                      0.07
2   2012           2.31             0.02                      0.06
3   2013           2.37             0.03                      0.08
4   2014           2.35             0.04                      0.08
5   2015           2.09             0.02                      0.06
6   2016           2.26             0.04                      0.09
7   2017           2.24             0.03                      0.07
8   2018           2.21             0.03                      0.08
9   2019           2.13             0.04                      0.10
10  2020           0.88             0.09                      0.33
11  2021           1.45             0.03                      0.11
12  2022           1.60             0.02                      0.09

The coefficient of variation for Tram journeys was low and stable from 2010 to 2019, spiked in 2020 due to the pandemic, and has been gradually decreasing since, approaching pre-pandemic levels by 2022.

In [39]:
SELECT 
    year,
    ROUND(AVG(journeys_millions), 2) AS mean_journeys,
    ROUND(VARIANCE(journeys_millions), 2) AS yearly_variance,
    ROUND(SQRT(VARIANCE(journeys_millions)) / AVG(journeys_millions), 2) AS coefficient_of_variation
FROM TFL.JOURNEYS
WHERE journey_type = 'Emirates Airline'
GROUP BY year, journey_type
ORDER BY year;

year  mean_journeys  yearly_variance  coefficient_of_variation
0   2010            NaN              NaN                       NaN
1   2011            NaN              NaN                       NaN
2   2012           0.21             0.02                      0.71
3   2013           0.12             0.00                      0.35
4   2014           0.12             0.00                      0.25
5   2015           0.12             0.00                      0.32
6   2016           0.12             0.00                      0.25
7   2017           0.11             0.00                      0.31
8   2018           0.11             0.00                      0.24
9   2019           0.10             0.00                      0.26
10  2020           0.03             0.00                      0.66
11  2021           0.11             0.00                      0.37
12  2022           0.12             0.00                      0.28

The data for Emirates Airline journeys shows a consistent pattern of zero yearly variance from 2013 to 2022, indicating no recorded fluctuation in journey counts. Despite this, the coefficient of variation remains relatively high, suggesting that even small changes in low mean values significantly impact perceived variability. This could point to data recording issues or minor variations not captured as significant variance.